### ISSUE TITLE
[Bug]: When using asset injection in some environments, core.min.js is missing closing );

### ISSUE THREAD

**Issue by nathan-io** [rappasoft/laravel-livewire-tables#1423]
### What happened?

In our app, we recently upgraded Laravel from 9 to 10, Livewire from 2 to 3, and `livewire-tables` to from 2.15.0 to v3.0-beta.3.

On my local (Ubuntu 20 on WSL2), our livewire-tables tables seem to be working fine.

However, on our staging instance (a Forge-provisioned server), as well as on one of our developer's local instances, we're seeing strange behavior:

On all pages, there are two new console errors:

```
core.min.js:3 Uncaught SyntaxError: missing ) after argument list (at core.min.js:3:6392)
livewire.js?id=178de384:7707 Uncaught SyntaxError: Invalid or unexpected token (at livewire.js?id=178de384:7707:1)
```

On pages which contains a livewire-table, there are [additional console errors](https://gyazo.com/9535ab05c5d356cf32056e3f5540cd37).

Visually, two things are happening:

- On tables where we're using filters, the filter button very briefly appears, but then disappears. You can't see or access the filters.
- On a different table where we don't have filters, [this area](https://gyazo.com/8481016bd73aab2074e058f172e5dc93) very briefly appears, then disappears.

---

I copied `core.min.js` from both instances. Diff [here](https://gyazo.com/bcada3484a2248c9aafe06baa814c7a5).

You can see the missing `);` in the version from staging. The version from my local has it.

This `core.min.js` response seems to come from `vendor/rappasoft/laravel-livewire-tables/resources/js/laravel-livewire-tables.min.js`.

When I look at that file, the missing `);` is there.

On our staging instance, I've tried:

1. `rm -rf vendor && composer install` 
2. `rm -rf node_modules && npm i && npm run build`
3. `php artisan view:clear && php artisan optimize:clear && php artisan clear`
4. Disabled Cloudflare cache
5. Restarted the entire server

I didn't think #3 was necessary, just wanted to be thorough.

I've also confirmed that both my local and staging have the same version of `laravel/framework`, `livewire/livewire`, and `rappasoft/livewire-tables`.

Despite this, the `core.min.js` on staging remains the same, even when I access the URL directly and do a hard refresh.

As noted above, this is also happening on the local instance of our developer. 

### How to reproduce the bug

Unsure how to reproduce. 

### Package Version

v3.0.0-beta.3

### PHP Version

8.1.x

### Laravel Version

10.28.0

### Alpine Version

3.12.0

### Theme

Tailwind 3.x

### Notes

It isn't something related to a particular branch. I've tried different branches on my local, and everything works as expected. 

Our Forge server is running Node 20.8.1 and npm 10.1.0. On my local, where everything works, Node is 18.2.0 and npm is v9.7.1. So I don't think it's related to the npm version.

**Comment by nathan-io** [comment]
As a follow up, our developer that is having this same issue on his machine is using npm 9.8.1 and node 18.18.0.

I tried on my laptop and everything works fine there too, using npm 9.6.7 and node 18.2.0.

I suppose this really isn't an issue with the package itself, but I'm not very experienced in this area and was just hoping that someone may have an insight as to why this is happening only on some instances of our app.

**Comment by nathan-io** [comment]
After realizing that `core.min.js` was being injected and looking into how that works, I think the issue lies with the output of the `response()->file()` in `pretendResponseIsJs()`.

We had an issue with this elsewhere in our app, when trying to return a PDF file as a response. The response was malformed. The solve for that was to add `ob_end_clean()`.

```
$response = response()->file($file);
ob_end_clean();
```

I got that solution from https://laracasts.com/discuss/channels/laravel/image-is-being-thrown-as-a-white-blank-image?reply=132045

It's possible that `response()->file()` has some quirks depending on the environment.

I've confirmed that even in the instances/environments where we were having the issue, `vendor/rappasoft/laravel-livewire-tables/resources/js/laravel-livewire-tables.min.js` is valid JS and includes the missing `);`. I think this is getting chopped off in some cases by `response->file()`.

---

We were able to bypass the issue by using bundler including, though we weren't able to get it working as documented [here](https://rappasoft.com/docs/laravel-livewire-tables/v3/start/including-assets#content-bundler-including) with Vite due to a postcss error.

Rather, we're importing like so:

common.js:

```
import '../../vendor/rappasoft/laravel-livewire-tables/resources/js/laravel-livewire-tables.min.js';
```

common.css:

```
@import '../../vendor/rappasoft/laravel-livewire-tables/resources/css/laravel-livewire-tables.min.css';
```

I'm trying to get more details on the error.

**Comment by lrljoe** [comment]
I'll take a look this evening, and will be around in the Rappasoft Discord

We reuse the livewire core injection capability, so it should just work.

Will have another look, which precise version of the package are you on beta1,beta2,beta3, or one of the dev branches?

**Comment by nathan-io** [comment]
Sounds good. We're on beta3.

**Comment by lrljoe** [comment]
Amazing, thanks for including the detailed error.

I'm looking at my dev version, and the line closure is definitely there, so I may have already fixed this and just merged it into a release just yet.

Given there's something not working properly, I'll get a beta4 out shortly.

**Comment by lrljoe** [comment]
I've got a native Ubuntu instance, and it doesn't get upset, when using a bundler approach, with:
```
import '../../vendor/rappasoft/laravel-livewire-tables/resources/imports/laravel-livewire-tables-all.js';
```
in my "app.js" (which is the root for my bundler)

I'll see if I can replicate your error on my end though!

**Comment by nathan-io** [comment]
> Are you using flatpickr elsewhere in your project out of interest?

I don't believe so.

Our `vite.config.js`:

```
import { defineConfig } from 'vite';
import laravel, { refreshPaths } from 'laravel-vite-plugin';

export default defineConfig({
    plugins: [
        laravel({
            input: [
                'resources/css/app.css',
                'resources/js/app.js',
                'resources/css/site.css',
                'resources/js/site.js',
            ],
            refresh: [
                ...refreshPaths,
                'app/Http/Livewire/**',
            ],
        }),
    ],
    resolve: {
        alias: {
            '@node': '/node_modules',
        },
    },
});
```

package.json

```
{
    "private": true,
    "scripts": {
        "dev": "vite",
        "build": "vite build"
    },
    "devDependencies": {
        "@alpinejs/focus": "^3.10.5",
        "@tailwindcss/forms": "^0.5.2",
        "@tailwindcss/typography": "^0.5.0",
        "alpinejs": "^3.12",
        "autoprefixer": "^10.4.7",
        "axios": "^0.27",
        "laravel-echo": "^1.15.2",
        "laravel-vite-plugin": "^0.8.0",
        "pusher-js": "^8.3.0",
        "tailwindcss": "^3.2.6",
        "vite": "^4.1.1",
        "postcss": "^8.4.14"
    },
    "dependencies": {
        "@alpinejs/collapse": "^3.12.0",
        "@alpinejs/focus": "^3.12.0",
        "@alpinejs/mask": "^3.12.0",
        "@alpinejs/ui": "^3.12.0-beta.0",
        "@caneara/iodine": "^8.3.0",
        "@splidejs/splide": "^4.1.4",
        "alpinejs": "^3.12.0",
        "intl-tel-input": "^17.0.21",
        "tippy.js": "^6.3.7"
    }
}
```


Is `core.min.js` contain the injection method work fine on your native Ubuntu machine?

**Comment by lrljoe** [comment]
I don't get that error appearing in any of my test/dev envs (which are all either Ubuntu or Debian), I'll try a fresh install and see if anything complains at that point.

**Comment by lrljoe** [comment]
Just ran some tests, and unable to replicate at the moment.

If you could please share initially your vite.config.js, I'll then try to replicate the issue on my end.

I'm sure you're probably doing something cleverer than I am, so it'd be good to try to replicate it properly.

**Comment by nathan-io** [comment]
Thanks but it's probably the other way around! It's in one of the comments above.

**Comment by lrljoe** [comment]
Ah k, spotted that vite config.

Couple of quick questions for you here.

Vite/Laravel both advise that you don't add CSS files as inputs, instead add them as imports into your JS.  It still separates them out properly.

E.g.
```
    plugins: [
        laravel({
            input: [
                'resources/js/app.js'
            ],
```
Then my app.js has:
```
import "../css/app.css";
```

Now I lost track of why exactly they recommend this, but I know it works okay for me!

Next up, you seem to be missing the common alias, that Laravel advise you add:

```
    resolve: {
        alias: {
            '@': '/resources/js'
        }
    }
```

Finally, I don't see any postcss etc going on there, so how are you building your Tailwind etc (minimally)

**Comment by lrljoe** [comment]
As an example of a vite config that I use for one of my demo environments:

```
import { defineConfig } from 'vite';
import laravel from 'laravel-vite-plugin';
import tailwindcss from 'tailwindcss';
import autoprefixer from 'autoprefixer';
import cssnano from 'cssnano';
import resolve from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';

export default defineConfig({
    build: {
        rollupOptions: {
            plugins: [
                resolve(),
                commonjs(),
            ]
        }
    },
    plugins: [
        laravel({
            input: [
                'resources/js/app.js'
            ],
            refresh: true,
            postcss: [
                tailwindcss(),
                autoprefixer(),
                cssnano({
                    preset: 'default',
                }),
            ],
        })
    ],
    resolve: {
        alias: {
            '@': '/resources/js'
        }
    }

});
```

With app.js containing:
```
import './bootstrap';
import { Livewire, Alpine } from '../../vendor/livewire/livewire/dist/livewire.esm';
import Clipboard from '@ryangjchandler/alpine-clipboard'
import focus from '@alpinejs/focus';
import '../../vendor/rappasoft/laravel-livewire-tables/resources/imports/laravel-livewire-tables-all';
import "../css/app.css";

Alpine.plugin(Clipboard)
Alpine.plugin(focus)
Livewire.start()
```

**Comment by nathan-io** [comment]
Thanks and sorry for the delay getting back to you! 

Our original frontend dev recently moved on from the project but I'll reach out to see why it was set up like that. I want to make sure we have a proper setup.

> Vite/Laravel both advise that you don't add CSS files as inputs, instead add them as imports into your JS.

I think I see what you're talking about in the Laravel docs:

> If you are building an SPA, including applications built using Inertia, Vite works best without CSS entry points:

Our app isn't really an SPA and we aren't using Interia, but maybe this is also a best practice for apps using Livewire?

As for the Vite docs, I couldn't find anything about this but this is admittedly not really my area. 

> Next up, you seem to be missing the common alias, that Laravel advise you add:

To my understanding this alias is defined by Laravel by default.

> I don't see any postcss etc going on there, so how are you building your Tailwind etc (minimally)

I _think_ this is working because we have a `postcss.config.js`:

```
module.exports = {
    plugins: {
        tailwindcss: {},
        autoprefixer: {},
    },
};
```

I've noticed that when I add a Tailwind class that we haven't used before, I have to rebuild assets before it is included.

**Comment by lrljoe** [comment]
@nathan-io - yep, that's how the tailwindcss/autoprefix works, it only extracts the classes you are using.

Using tailwind.config.js you can tell it specific classes to always add, or content to include from 3rd parties using purge.content and purge.safelist

**Comment by lrljoe** [comment]
@nathan-io - are you still experiencing this issue?  If so, please respond to re-open the issue.

**Comment by nathan-io** [comment]
As to the original issue, I don't know because we switched from injection to bundling in order to bypass it. I think this is something that others could see as well as `response()->file()` doesn't seem to behave consistently across environments. 

https://github.com/rappasoft/laravel-livewire-tables/issues/1423#issuecomment-1766668468

The issue where we had to tweak the asset bundling could be related to the specifics of our Vite setup, I'm not sure.
